packages/react/menu/src/Menu.tsx

```diff
@@ -456,6 +459,13 @@ const MenuContentImpl = React.forwardRef((props, forwardedRef) => {
             contentRef.current?.focus();
             setCurrentItemId(null);
           }, [])}
+          onPointerGraceAreaChange={React.useCallback(
+            (area) => (pointerGraceAreaRef.current = area),
+            []
+          )}
+          isMouseMovingToSubmenu={(event) =>
```

Comment:
I was expecting this to be named something like `onMouseMoveToSubmenu` and then I saw it returns a boolean. We don't usually do this so I tried to work out why we're suddenly needing to now...

It feels like another case where we might be better off just passing the ref through context and calling the `isMouseMovingToSubMenu` utility inline. Or preferably something like this:

```jsx
// new handler fired from `onMouseMove`
onItemEnter={React.useCallback((event) => {
	if (isMouseMovingToSubmenu(event, pointerGraceAreaRef.current)) {
		event.preventDefault();
	}
}, []))
onItemLeave={React.useCallback((event) => {
	if (isMouseMovingToSubmenu(event, pointerGraceAreaRef.current)) {
		event.preventDefault();
	}
}, [])}
```

This reads more like English to me, like, when an item is entered, if we're moving to submenu then do nothing and vice versa.

I'm not sure if that would work though. Thoughts?

Reply:
That's exactly what we had before we changed it. But it was named differently. I'm easy on that one, with the new name for `pointerGraceAreaRef` it does read better.

Reply:
Which did you have before, the `event.preventDefault()` stuff or passed the ref through context? I think I prefer the `event.preventDefault()` approach personally but happy with either or.

Reply:
Oh didn't realise you had new handlers, I meant the ref through context. I don't really get why we'd need to re-create `onMouseEnter`/`onMouseLeave`…

Reply:
We already have an `onItemLeave` so that's not a new thing, only `onItemEnter` would exist which would replace the strange `isMouseMovingToSubmenu` method. It's not new code, just a rename and it would `event.preventDefault()` instead of returning a boolean.

Reply:
Yeah but that's quite different, if I follow correctly, you're talking about the current `onItemLeave` on context?
If so, that's just a void callback, not a custom event. We'd have to implement 2 new custom events to do the prevent default thing.

Reply:
> you're talking about the current `onItemLeave` on context

yes, on the same context where that `isMouseMovingToSubmenu` method is.

We wouldn't have to add custom events... `onItemLeave` is called from existing events. It's just a case of passing that event to it:

```jsx
onItemEnter(event: React.MouseEvent): void;
onItemLeave(event: React.MouseEvent): void;
```

![image](https://user-images.githubusercontent.com/175330/120845800-32d52700-c569-11eb-8402-b7fa1ae0ca71.png)

Seems to work okay but you'll know better than me 😅

Reply:
This made me realise we could even do this then which feels more "complete":

![image](https://user-images.githubusercontent.com/175330/120848593-e390f580-c56c-11eb-94a1-1ffd5734ca4b.png)

So our two `onItemEnter` and `onItemLeave` events handle the side effects:

![image](https://user-images.githubusercontent.com/175330/120848659-f60b2f00-c56c-11eb-99a7-eed5a6c3a7c4.png)